### PR TITLE
Post function

### PR DESCRIPTION
**Pull request by cammac60**
- This PR adds event delegation for our form section and send a post request whenever data is submitted. 

--------------

- A class was added to all form inputs in `index.html` and the post method was removed from all form tags - starting on line 45.
- The event handler and post methods start on line 159 of `index.js`

--------------

TO DO: 

- We still need a bit of form validation added to make sure users can't submit blank info. 
- We still need proper error handling.
- We need to do some more solid testing to make sure the post request is actually going through. 
- I had a ton of issues with merge conflicts and version control. For some reason this push brought a lot of old commits along with it. Everything still works fine on my machine but please pull this branch and make sure there are no glaring issues before you merge.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -42,45 +42,45 @@ <h3>
...
-          <input aria-required="true" id="form__control--steps" type="number" name="number" pattern="/^[0-9]+$/" min="1" max="20000" placeholder="#">
+          <input class="form-control" aria-required="true" id="form__control--steps" type="number" name="number" pattern="/^[0-9]+$/" min="1" max="20000" placeholder="#">
           </label>
           <label for="form__control--minutes">Minutes Active
-          <input aria-required="true" id="form__control--minutes" type="number" name="number" pattern="/^[0-9]+$/" min="1.0" max="400" placeholder="#">
+          <input class="form-control" aria-required="true" id="form__control--minutes" type="number" name="number" pattern="/^[0-9]+$/" min="1.0" max="400" placeholder="#">
           </label>
           <label for="form__control--stairs">Flights of Stairs
-          <input aria-required="true" id="form__control--stairs" type="number" name="number" pattern="/^[0-9]+$/" min="1.0" max="500" placeholder="#">
+          <input class="form-control" aria-required="true" id="form__control--stairs" type="number" name="number" pattern="/^[0-9]+$/" min="1.0" max="500" placeholder="#">
```
Comment (trishalanglois) [inline src/index.html:56]: The pattern you put in for all the forms is great, Cameron!  When I pulled this down and tested it, I couldn't even enter a letter.

File: src/index.js [hunk 2]
```diff
@@ -156,6 +156,64 @@ const stepTrends = $('#step-trends');
...
+      body: JSON.stringify({
+        userID: currentUserID,
+        date: $('#form__control--date1').val(),
+        numSteps: parseInt($('#form__control--steps').val()),
+        minutesActive: parseInt($('#form__control--minutes').val()),
+        flightsOfStairs: parseInt($('#form__control--stairs').val())
+      })
+    })
+    .then(response => response.json())
+    .then(data => console.log(data))
```
Comment (trishalanglois) [inline src/index.js:176]: This is super helpful to see that the response is happening, but we should probably take this out before submission.

**Comment by trishalanglois** [comment]
I pulled it down, and it seems like overall functionality is working.  Nice work, Cameron!

I made an issue for the error handling, as we need that for the splash page as well.  I think the main To-Do that we should include before we merge this is the form validating piece.  Cameron, the pattern attr and min length requirements are both resulting in the .catch error messages in the console for me, which is great.  What did you have in mind for your first bullet point about users not submitting blank info -- just an error message being displayed on the DOM?  (Because the server is already giving us an error message if we leave any required fields blank --yay!).

I know a lot of troubleshooting went into this PR, so kudos to you and your perseverance 👊

File: src/index.html [hunk 3]
```diff
@@ -42,45 +42,45 @@ <h3>
...
+            <input class="form-control" aria-required="true" id="form__control--water" type="number" name="number" pattern="[-+]?[0-9]*[.,]?[0-9]+" step=0.01 min="1" max="200" placeholder="#">
             </label>
-            <button type="submit">Submit</button>
+            <button id="hydration-submit" type="submit">Submit</button>
           </form>
       </article>
       <article class="addSleep__article">
-        <form method="POST" >
+        <form>
             <label for="form__control--date3">Date
```
Comment (allisonjw) [inline src/index.html:74]: Good idea to remove the POST method. I didn't have it set up correctly to work and I don't think it is a good use case for this project.

**Comment by allisonjw** [comment]
@cammac60 I agree we need some additional form control. I tried to find the correct regex pattern for the yyyy/dd/mm format to insert into HTML. But I think this may be a good use for a function to test the date format is correct. Well done with the post method. It's easy to read and works great!

File: src/index.js (hunk 2 above)
Comment (cammac60) [reply src/index.js:176]: I agree, I was going to add on to this when I do some error handling.

**Comment by cammac60** [comment]
@trishalanglois I think we can probably use our current error messages when the user doesn't fill out a field. Allison set those up so I'll have to look at it a little more to get the message to trigger when a user tries to submit.

**Comment by cammac60** [comment]
@allisonjw I have a few ideas of how we can make that a little easier since I'm sure the regex for that is a bit complicated. I'll do some research and see if I can find anything else!

**Pull request merged**